pkg/gardenlet/controller/factory.go

```diff
@@ -0,0 +1,149 @@
+	backupentrycontroller "github.com/gardener/gardener/pkg/gardenlet/controller/backupentry"
+	controllerinstallationcontroller "github.com/gardener/gardener/pkg/gardenlet/controller/controllerinstallation"
+	seedcontroller "github.com/gardener/gardener/pkg/gardenlet/controller/seed"
+	shootcontroller "github.com/gardener/gardener/pkg/gardenlet/controller/shoot"
+	"github.com/gardener/gardener/pkg/logger"
+	"github.com/gardener/gardener/pkg/operation/common"
+	"github.com/gardener/gardener/pkg/operation/garden"
+	"github.com/gardener/gardener/pkg/utils/imagevector"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/version"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	kubeinformers "k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+)
+
+// GardenControllerFactory contains information relevant to controllers for the Garden API group.
+type GardenControllerFactory struct {
+	cfg                    *config.GardenletConfiguration
+	identity               *gardencorev1alpha1.Gardener
+	gardenNamespace        string
+	k8sGardenClient        kubernetes.Interface
+	k8sGardenCoreInformers gardencoreinformers.SharedInformerFactory
+	k8sInformers           kubeinformers.SharedInformerFactory
+	recorder               record.EventRecorder
+}
+
+// NewGardenControllerFactory creates a new factory for controllers for the Garden API group.
+func NewGardenControllerFactory(k8sGardenClient kubernetes.Interface, gardenCoreInformerFactory gardencoreinformers.SharedInformerFactory, kubeInformerFactory kubeinformers.SharedInformerFactory, cfg *config.GardenletConfiguration, identity *gardencorev1alpha1.Gardener, gardenNamespace string, recorder record.EventRecorder) *GardenControllerFactory {
+	return &GardenControllerFactory{
+		cfg:                    cfg,
+		identity:               identity,
+		gardenNamespace:        gardenNamespace,
+		k8sGardenClient:        k8sGardenClient,
+		k8sGardenCoreInformers: gardenCoreInformerFactory,
+		k8sInformers:           kubeInformerFactory,
+		recorder:               recorder,
+	}
+}
+
+// Run starts all the controllers for the Garden API group. It also performs bootstrapping tasks.
+func (f *GardenControllerFactory) Run(ctx context.Context) {
+	var (
+		// Garden core informers
+		backupBucketInformer           = f.k8sGardenCoreInformers.Core().V1alpha1().BackupBuckets().Informer()
+		backupEntryInformer            = f.k8sGardenCoreInformers.Core().V1alpha1().BackupEntries().Informer()
+		cloudProfileInformer           = f.k8sGardenCoreInformers.Core().V1alpha1().CloudProfiles().Informer()
+		controllerRegistrationInformer = f.k8sGardenCoreInformers.Core().V1alpha1().ControllerRegistrations().Informer()
+		controllerInstallationInformer = f.k8sGardenCoreInformers.Core().V1alpha1().ControllerInstallations().Informer()
```

Comment:
As explained above, should be from seed.
```suggestion
		controllerInstallationInformer = f.k8sSeedInformers.core().V1alpha1().ControllerInstallations().Informer()
```

Reply:
As explained above: not now :)